hoot-core-test/src/test/cpp/hoot/core/algorithms/splitter/CornerSplitterTest.cpp

```diff
@@ -107,6 +111,28 @@ class CornerSplitterTest : public HootTestFixture
                      "test-files/conflate/splitter/CornerSplitterRoundedExpected.osm");
   }
 
+  void runDogLegTest()
+  {
+    OsmXmlReader reader;
+
+    OsmMapPtr map(new OsmMap());
+    reader.setDefaultStatus(Status::Unknown1);
+    reader.read("test-files/conflate/splitter/CornerSplitterDogLeg.osm", map);
+
+    MapProjector::projectToPlanar(map);
+
+    CornerSplitter::splitCorners(map);
+
+    MapProjector::projectToWgs84(map);
+
+    OsmXmlWriter writer;
+    writer.setIncludeCompatibilityTags(false);
+    writer.write(map, "test-output/conflate/splitter/CornerSplitterDogLegOut.osm");
+
+    HOOT_FILE_EQUALS("test-output/conflate/splitter/CornerSplitterDogLegOut.osm",
+                     "test-files/conflate/splitter/CornerSplitterDogLegExpected.osm");
+  }
+
```

Comment:
This dog-leg test runs the splitter on four different ways to ensure that the small (less than circular error) dog-legs aren't split off by themselves.